dbms/src/WindowFunctions/tests/gtest_window_functions.cpp

```diff
@@ -0,0 +1,364 @@
+            }
+            else
+            {
+                vec.push_back({});
+            }
+        }
+        DataTypePtr data_type = makeNullable(std::make_shared<DataTypeMyDateTime>(fsp));
+        return {makeColumn<Nullable<MyDateTime>>(data_type, vec), data_type, name, 0};
+    }
+
+    void setMaxBlockSize(int size)
+    {
+        context.getSettingsRef().max_block_size.set(size);
+    }
+
+    void mockInterpreter(std::vector<NameAndTypePair> source_columns, Context context)
+    {
+        std::vector<BlockInputStreams> mock_input_streams_vec = {};
+        DAGQueryBlock mock_query_block(0, static_cast<const google::protobuf::RepeatedPtrField<tipb::Executor>>(nullptr));
+        std::vector<SubqueriesForSets> mock_subqueries_for_sets = {};
+        mock_interpreter = std::make_shared<DAGQueryBlockInterpreter>(context,
+                                                                      mock_input_streams_vec,
+                                                                      mock_query_block,
+                                                                      1);
+
+        mock_interpreter->analyzer = std::make_unique<DAGExpressionAnalyzer>(std::move(source_columns), context);
+    }
+
+    void mockExecuteTableScan(DAGPipeline & pipeline, ColumnsWithTypeAndName columns)
+    {
+        pipeline.streams.push_back(std::make_shared<MockTableScanBlockInputStream>(columns, context.getSettingsRef().max_block_size));
+        mock_interpreter->input_streams_vec.push_back(pipeline.streams);
+    }
+
+    void mockExecuteWindowOrder(DAGPipeline & pipeline, std::string sort_json_str)
+    {
+        tipb::Sort sort;
+        google::protobuf::util::JsonStringToMessage(sort_json_str, &sort);
+        mock_interpreter->handleWindowOrder(pipeline, sort);
+        mock_interpreter->input_streams_vec[0] = pipeline.streams;
+        NamesWithAliases final_project;
+        for (const auto & column : (*mock_interpreter->analyzer).source_columns)
+        {
+            final_project.push_back({column.name, ""});
+        }
+        mockExecuteProject(pipeline, final_project);
+    }
+
+    void mockExecuteWindow(DAGPipeline & pipeline, std::string window_json_str)
+    {
+        tipb::Window window;
+        google::protobuf::util::JsonStringToMessage(window_json_str, &window);
+        mock_interpreter->handleWindow(pipeline, window);
+        mock_interpreter->input_streams_vec[0] = pipeline.streams;
+        NamesWithAliases final_project;
+        for (const auto & column : (*mock_interpreter->analyzer).source_columns)
+        {
+            final_project.push_back({column.name, ""});
+        }
+        mockExecuteProject(pipeline, final_project);
+    }
+
+    void mockExecuteProject(DAGPipeline & pipeline, NamesWithAliases & final_project)
+    {
+        mock_interpreter->executeProject(pipeline, final_project);
+    }
+
+    static void checkBlock(const Block & lhs, const Block & rhs)
```

Comment:
Maybe should move this function to `FunctionTestUtils.cpp`, so it can be used by others.

Reply:
> Maybe should move this function to `FunctionTestUtils.cpp`, so it can be used by others.

done.